### PR TITLE
docs: add memory and swap memory usage examples

### LINKED ISSUE #11629: Document the relationship between --memory and --memory-swap
**Problem**
 
There are two flags --memory and --memory-swap and the interaction between both of them is unclear in the documentation
 
**Solution**
 
Add documentation for using --memory and --memory-swap together and how they relate and interact with each other.  i.e. does the memory limit have to be larger or smaller than the swap limit?

### PR DESCRIPTION
**Pull request by hqhq**
fix: https://github.com/docker/docker/issues/11629

Signed-off-by: Qiang Huang h.huangqiang@huawei.com

### CODE REVIEW

File: docs/sources/reference/run.md [hunk 1]
```diff
@@ -507,6 +507,29 @@ We have four ways to set memory usage:
   </tbody>
 </table>
 
+Examples:
+
+    # docker run -ti ubuntu:14.04 /bin/bash
```
Comment (SvenDowideit) [inline docs/sources/reference/run.md]: atm we use `$ sudo docker run ...` for out examples :)

File: docs/sources/reference/run.md [hunk 2]
```diff
@@ -507,6 +507,29 @@ We have four ways to set memory usage:
...
+
+We set nothing about memory, this means we can use as much memory and swap memory
```
Comment (SvenDowideit) [inline docs/sources/reference/run.md]: perhaps replace `we` with `the processes in the container`

**Comment by SvenDowideit** [comment]
@hqhq - very awesome again, thank you! 

LGTM with some language tweaks @docker/docs-owners

File: docs/sources/reference/run.md [hunk 3]
```diff
@@ -507,6 +507,29 @@ We have four ways to set memory usage:
...
+    # docker run -ti -m 300M --memory-swap -1 ubuntu:14.04 /bin/bash
+
+We set memory limit and disabled swap memory limit, this means we can use 300M
+memory and as much swap memory as we need (if the host supports swap memory).
+
+    # docker run -ti -m 300M ubuntu:14.04 /bin/bash
+
+We set memory limit only, this means we can use 300M memory and 300M swap memory,
+by default, memory-swap will be set as double of memory, in this case, memory + swap
+would be 2*300M, so we can use 300M swap memory as well.
```
Comment (SvenDowideit) [inline docs/sources/reference/run.md]: mmm, something needs tweaking here - I think perhaps changing the `memory-swap will be set to double of memory` makes it less clear that we're trying to say that the `--memory-swap` flag is the total virtual memory size, including the `--memory` portion.

**Comment by hqhq** [comment]
Thanks @SvenDowideit , updated.

File: docs/sources/reference/run.md (hunk 3 above)
Comment (hqhq) [inline docs/sources/reference/run.md]: At the beginning of section, we already told that `--memory-swap` is the total memory, including memory and swap. So I think readers won't confuse that bit. I changed that sentence a little, hope it can be clearer.

**Comment by hqhq** [comment]
@SvenDowideit assumes we put details and examples in run.md, I added cpuset in this doc too. Some other resource configs support are on going, their usage details and examples can all be put in this section.
Please review again, thanks.

**Comment by crosbymichael** [comment]
LGTM

ping @moxiegirl

File: docs/sources/reference/run.md [hunk 4]
```diff
@@ -426,36 +426,37 @@ the `--security-opt` flag. For example, you can specify the MCS/MLS level, a
 requirement for MLS systems. Specifying the level in the following command
 allows you to share the same content between containers.
 
-    # docker run --security-opt label:level:s0:c100,c200 -i -t fedora bash
+    $ sudo docker run --security-opt label:level:s0:c100,c200 -i -t fedora bash
```
Comment (Sirupsen) [inline docs/sources/reference/run.md]: Why was this changed from being implicit with `#` to `sudo`? Is that consistent with the rest of the docs?

Comment (hqhq) [inline docs/sources/reference/run.md]: That's what @SvenDowideit suggested, atm we use `$ sudo docker run ...` for our examples, I haven't check other docs, maybe we can have a PR to change all of them.

Comment (Sirupsen) [inline docs/sources/reference/run.md]: Ah, I missed the comment since it was folded

**Comment by moxiegirl** [comment]
@hqhq @Sirupsen Actually, we went through and removed all the sudo from our examples.  Why? Because now on Ubuntu and other Linux systems we ask people to create a docker group and add docker to it.  On OSX, actually, you should not use sudo.   That said, I'll fix this up myself and chat to Sven also!

Thank you for your contribution @hqhq Good job. LGTM
